src/Tracing.cpp

```diff
@@ -236,15 +236,17 @@ class InjectTracing : public IRMutator {
                 builder.coordinates.push_back(extent);
             }
 
-            builder.event = (op->is_producer ?
+            const bool is_producer = op->is_producer;
+            builder.event = (is_producer ?
                              halide_trace_end_produce :
                              halide_trace_end_consume);
             Expr call = builder.build();
             Stmt new_body = Block::make(op->body, Evaluate::make(call));
 
-            stmt = ProducerConsumer::make(op->name, op->is_producer, new_body);
+            // Note: this call makes op invalid; don't reference it later
+            stmt = ProducerConsumer::make(op->name, is_producer, new_body);
```

Comment:
It will also be safe, and easier to upgrade to returning the new Stmt, if we just making assigning to stmt/expr the last thing we do. I.e:

    Stmt new_body = ProducerConsumer::make(op->name, op->is_producer, new_body);
    ...
    stmt = LetStmt::make(f.name() + ".trace_id", call, new_body);